src/segmentedfilesystem/ReceivedFile.java

```diff
@@ -0,0 +1,79 @@
+package segmentedfilesystem;
+
+import java.io.*;
+import java.util.*;
+
+public class ReceivedFile {
+    byte fileID;
+    TreeMap<Integer, DataPacket> dataPackets;
+    HeaderPacket header;
+    int totalDataPackets;
+
+    public ReceivedFile(byte fileID, Packet packet) {
+        this.fileID = fileID;
+        dataPackets = new TreeMap<Integer, DataPacket>();
+        header = null;
+        totalDataPackets = -1;
+
+        this.addPacket(packet);
+    }
+
+    public void addPacket(Packet packet) {
+        if (packet.isHeaderPacket() == true) {
+            this.header = (HeaderPacket) packet;
```

Comment:
If you'd had a `addHeaderPacket()` with the appropriate signature, you could call it and avoid the explicit cast. Ditto for `addDataPacket()`.